lib/pure/asyncdispatch.nim

```diff
@@ -1986,6 +2010,62 @@ proc send*(socket: AsyncFD, data: string,
 import asyncmacro
 export asyncmacro
 
+proc doAfter*(ms: int or float, todo: proc ()): PendingOps =
+  ## Executes actions passed as `todo` after `ms` milliseconds
+  ## Without blocking the main execution flow while waiting
+  ## An equivalent of javascript's setTimeout
+
+  runnableExamples:
+    discard doAfter(2_500) do():
+      echo "2.5 seconds passed !"
+
+    var pend = doAfter(3_000) do():
+      echo "This line will never be executed !"
+
+    # Let's cancel the second pennding process
+    # 1.5 seconds before its execution :
+    discard doAfter(1_500) do(): cancel pend
+
+  var pend = PendingOps()
```

Comment:
Why not `result`? (same for cycle in doEvery)

Reply:
You're right. It's because the first implementation used templates. I'm going to change it.

Reply:
Finally, in this context, `result` is not the best. Because the inner proc copies a reference to the object. I can't write in the inner proc: `var pend = outerProc.result`.